applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -230,6 +234,239 @@ void CopyNodalSolutionStepVariablesList(
+        r_condition.SetValue(RANS_IS_OUTLET, r_condition.Is(OUTLET));
+        r_condition.SetValue(RANS_IS_STRUCTURE, r_condition.Is(STRUCTURE));
+    }
+}
+
+void FixFlaggedDofs(ModelPart& rModelPart,
+                    const Variable<double>& rFixingVariable,
+                    const Flags& rFlag,
+                    const bool CheckValue)
+{
+    KRATOS_TRY
+
+    const int number_of_nodes = rModelPart.NumberOfNodes();
+#pragma omp parallel for
+    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+    {
+        auto& r_node = *(rModelPart.NodesBegin() + i_node);
+        if (r_node.Is(rFlag) == CheckValue)
+        {
+            r_node.Fix(rFixingVariable);
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+void CopyNodalFlaggedVariableFromNonHistoricalToHistorical(ModelPart& rModelPart,
+                                                           const Variable<TDataType>& rVariable,
+                                                           const Flags& rFlag,
+                                                           const bool CheckValue)
+{
+    KRATOS_TRY
+
+    const int number_of_nodes = rModelPart.NumberOfNodes();
+#pragma omp parallel for
+    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+    {
+        auto& r_node = *(rModelPart.NodesBegin() + i_node);
+        if (r_node.Is(rFlag) == CheckValue)
+        {
+            r_node.FastGetSolutionStepValue(rVariable) = r_node.GetValue(rVariable);
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+void CopyNodalFlaggedVariableFromHistoricalToNonHistorical(ModelPart& rModelPart,
+                                                           const Variable<TDataType>& rVariable,
+                                                           const Flags& rFlag,
+                                                           const bool CheckValue)
+{
+    KRATOS_TRY
+
+    const int number_of_nodes = rModelPart.NumberOfNodes();
+#pragma omp parallel for
+    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+    {
+        auto& r_node = *(rModelPart.NodesBegin() + i_node);
+        if (r_node.Is(rFlag) == CheckValue)
+        {
+            r_node.SetValue(rVariable, r_node.FastGetSolutionStepValue(rVariable));
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+void CalculateMagnitudeSquareForNodal3DVariable(ModelPart& rModelPart,
+                                                const Variable<array_1d<double, 3>>& r3DVariable,
+                                                const Variable<double>& rOutputVariable)
+{
+    const int number_of_nodes = rModelPart.NumberOfNodes();
+#pragma omp parallel for
+    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+    {
+        auto& r_node = *(rModelPart.NodesBegin() + i_node);
+        const double magnitude = norm_2(r_node.FastGetSolutionStepValue(r3DVariable));
+        r_node.FastGetSolutionStepValue(rOutputVariable) = std::pow(magnitude, 2);
```

Comment:
Do you require the output variable to be historical?

Reply:
It is not a must, since I use an existing variable (VELOCITY_POTENTIAL), it does not have an additional cost. This is a specific method required only by incompressible potential flow solver.